File: src/main/java/com/leijendary/spring/notificationtemplate/config/PropertiesConfiguration.java
package com.leijendary.spring.notificationtemplate.config;

import com.leijendary.spring.notificationtemplate.config.properties.*;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableConfigurationProperties({
        AspectProperties.class,
        AuthProperties.class,
        CorsProperties.class,
        InfoProperties.class,
        SmsProperties.class })
public class PropertiesConfiguration {
}

File: src/main/kotlin/com/leijendary/spring/template/notification/core/config/PropertiesConfiguration.kt
package com.leijendary.spring.template.notification.core.config

import com.leijendary.spring.template.notification.core.config.properties.*
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Configuration

@Configuration
@EnableConfigurationProperties(
    AuthProperties::class,
    AwsSnsProperties::class,
    DataSourcePrimaryProperties::class,
    DataSourceReadonlyProperties::class,
    InfoProperties::class,
    KafkaTopicProperties::class,
    TwilioProperties::class
)
class PropertiesConfiguration 
